src/Sylius/Bundle/ShopBundle/EventListener/OrderCustomerIpListener.php

```diff
@@ -30,6 +30,13 @@ public function assignCustomerIpToOrder(GenericEvent $event): void
         $subject = $event->getSubject();
         Assert::isInstanceOf($subject, OrderInterface::class);
 
-        $this->ipAssigner->assign($subject, $this->requestStack->getMasterRequest());
+        if (method_exists($this->requestStack, 'getMainRequest')) {
+            $request = $this->requestStack->getMainRequest();
+        } else {
+            $request = $this->requestStack->getMasterRequest();
+        }
+        Assert::notNull($request);
```

Comment:
Please add some error message, webmozart's `notNull` asserts a notorious hard to debug, when stack trace points to vendor/webmozaert/assert/assert and not to offending line of code.

Reply:
I agree it's always painful 😉